test/e2e/clusterctl_upgrade.go

```diff
@@ -71,6 +71,24 @@ type ClusterctlUpgradeSpecInput struct {
 	// InitWithKubernetesVersion can be used to override the INIT_WITH_KUBERNETES_VERSION e2e config variable with a specific
 	// Kubernetes version to use to create the secondary management cluster, e.g. `v1.25.0`
 	InitWithKubernetesVersion string
+	// InitWithCoreProvider can be used to override the INIT_WITH_CORE_PROVIDER e2e config variable with a specific core provider
+	// version to use when initializing the secondary management cluster, e.g. `cluster-api:v1.3.0`
+	InitWithCoreProvider string
+	// InitWithBootstrapProviders can be used to override the INIT_WITH_BOOTSTRAP_PROVIDERS e2e config variable with specific
+	// bootstrap providers and versions to use when initializing the secondary management cluster, e.g. `kubeadm:v1.3.0`
```

Comment:
on top of the comment above "bootstrap providers and versions" doesn't seem to fit here, probably should be "bootstrap provider versions"
(same below)